src/poll.iops.cpp

```diff
@@ -23,18 +23,14 @@ struct felspar::io::poll_warden::sleep_completion : public completion<void> {
         handle = h;
         timer = posix::fd{::timerfd_create(CLOCK_MONOTONIC, TFD_NONBLOCK)};
         if (not timer) {
-            exception =
-                    std::make_exception_ptr(felspar::stdexcept::system_error{
-                            errno, std::generic_category(), "timerfd_create",
-                            std::move(loc)});
+            error.code = {errno, std::system_category()};
+            error.message = "timerfd_create";
```

Comment:
With all these assignments, perhaps it becomes worth it to construct `error` at the call site and move-assign it into place?

Something like:
```cxx
    error = error_store{errno, std::system_category{}, "timerfd_create"};
```

And I suppose you could set the `source_location` as a parameter-with-default-value to that constructor.

Reply:
I've implement a common `outcome` type now, which I like, but doesn't address the original note you made :-) But I have an idea for that. As I hoped the `outcome` type has managed to subsume all of the differences I had in various `<void>` specialisations, so that's greatly simplified a lot of the rest of the code.

Reply:
And that should take care of this